test-e2e/assertions.ts

```diff
@@ -0,0 +1,92 @@
+
+const getTitleOfWindow = (app: ElectronApplication, page: Page) =>
+    new Promise(resolve =>
+        app
+            .browserWindow(page)
+            .then(browserWindow => browserWindow.getProperty('title'))
+            .then(property => resolve(property.jsonValue()))
+    );
+
+export const checkTitleOfWindow = async (
+    app: ElectronApplication,
+    titleString: string
+) => {
+    const window = await app.firstWindow();
+    const title = await getTitleOfWindow(app, window);
+    expect(title).toContain(titleString);
+};
+
+export const checkTitleOfSecondWindow = async (
+    app: ElectronApplication,
+    titleString: string
+) => {
+    const windows = await app.windows();
+    await expect(
+        getTitleOfWindow(app, windows[windows.length - 1])
+    ).resolves.toContain(titleString);
+};
+
+export const checkAppListContains = async (page: Page, appName: string) => {
+    await page.waitForSelector('.list-group-item');
+    await expect(
+        page.$$eval('.list-group-item .h8', apps =>
+            apps.map(app => app.textContent)
+        )
+    ).resolves.toContain(appName);
+};
+
+export const checkHasInstallButton = async (page: Page, appName: string) => {
+    await page.waitForSelector('.list-group-item');
+    await expect(
+        page.$(`[title="Install ${appName}"]`)
+    ).resolves.not.toBeNull();
+};
+
+export const checkHasNoInstallButton = async (page: Page, appName: string) => {
+    await page.waitForSelector('.list-group-item');
+    await expect(page.$(`[title="Install ${appName}"]`)).resolves.toBeNull();
+};
+
+export const checkHasNoLaunchButton = async (page: Page, appName: string) => {
+    await page.waitForSelector('.list-group-item');
+    await expect(page.$(`[title="Open ${appName}"]`)).resolves.toBeNull();
+};
+
+export const checkShowsAppUpdate = async (
+    page: Page,
+    appName: string,
+    availableVersion: string
+) => {
+    await page.waitForSelector(`[title="Update ${appName}"]`);
+    expect(page.locator(`.list-group-item`)).toContainText(
+        `${availableVersion} available`
+    );
+};
+
+export const checkShowsNoAppUpdate = async (page: Page, appName: string) => {
+    await page.waitForSelector('.list-group-item');
+    await expect(page.$(`[title="Update ${appName}"]`)).resolves.toBeNull();
+    await expect(
+        page.$(`.list-group-item:has-text("available")`)
+    ).resolves.toBeNull();
+};
+
+export const checkHasRemoveButton = async (page: Page, appName: string) => {
+    await page.waitForSelector('.list-group-item');
+    await page.click('.list-group-item button[aria-haspopup="true"]');
+    await expect(page.$(`[title="Remove ${appName}"]`)).resolves.not.toBeNull();
+};
+
+export const checkHasNoRemoveButton = async (page: Page, appName: string) => {
+    await page.waitForSelector('.list-group-item');
+    await page.click('.list-group-item button[aria-haspopup="true"]');
+    await expect(page.$(`[title="Remove ${appName}"]`)).resolves.toBeNull();
+};
```

Comment:
I don't know if this is feasible, but I generally prefer tests to test what the users actually see, and not implementation details. For example, instead of clicking `await page.click('.list-group-item button[aria-haspopup="true"]');`, which is not very readable and also more brittle (e.g. the class name might change or the aria attribute), it would be nicer to use something like `await page.click('Toggle ${appName} dropdown')`. Now in this particular instance, that isn't so easy, since the button doesn't have a name, only a caret symbol, so then it could be useful to add a test-id to the element, which doesn't interfere with normal use, for example: `await page.click('[data-test="toggle-${appName}-dropdown"]');` or something similar.